services/storage/store.go

```diff
@@ -385,53 +462,159 @@ func (s *Store) TagValues(ctx context.Context, req *datatypes.TagValuesRequest)
 	return cursors.NewStringSliceIterator(names), nil
 }
 
-type MeasurementNamesRequest struct {
-	MeasurementsSource *types.Any
-	Predicate          *datatypes.Predicate
+func (s *Store) MeasurementNames(ctx context.Context, mqAttrs *metaqueryAttributes) (cursors.StringIterator, error) {
+	if mqAttrs.pred != nil {
```

Comment:
`show measurements` test?

Reply:
Confirmed that InfluxQL is unaffected by this change, adding a `panic` doesn't cause any existing tests to fail